fhir-server/src/main/java/com/ibm/fhir/server/util/IssueTypeToHttpStatusMapper.java

```diff
@@ -36,15 +43,20 @@
             for (Issue issue : issues) {
                 IssueType code = issue.getCode();
                 if (code != null && code.getValue() != null) {
-                    return issueTypeToResponseCode(IssueType.ValueSet.from(code.getValue()));
+                    ValueSet issueType = IssueType.ValueSet.from(code.getValue());
+                    // Special case for IssueType CONFLICT which can be either an HTTP 409 (Conflict) or HTTP 412 (Precondition failed)
+                    if (issueType == IssueType.ValueSet.CONFLICT &&
+                            FHIRUtil.getExtensionStringValue(code, EXTENSION_URL_HTTP_FAILED_PRECONDITION) != null) {
```

Comment:
is this method used anywhere else? is it in FHIRUtil because you think it will be? it might be more containable to add it to this class.

Reply:
FHIRUtil already had `getExtensionStringValue(Resource, String)` so I basically just added a sister one for Elements.  I think both could be modified to be more generic, but at least it won't throw a ClassCastException any more when you specify an extension url with a non-string value ;-)